x-pack/plugins/index_management/public/hooks/use_details_page_mappings_model_management.ts

```diff
@@ -27,20 +28,18 @@ const getCustomInferenceIdMap = (
 ) => {
   return models?.data.reduce<InferenceToModelIdMap>((inferenceMap, model) => {
     const inferenceId = model.model_id;
-    const trainedModelId =
-      'model_id' in model.service_settings ? model.service_settings.model_id : '';
-
+    const trainedModelId = 'model_id' in model.service_settings && (model.service_settings.model_id == ElasticsearchModelDefaultOptions.elser || model.service_settings.model_id == ElasticsearchModelDefaultOptions.e5)?model.service_settings.model_id: undefined
     inferenceMap[inferenceId] = {
       trainedModelId,
-      isDeployable: model.service === 'elser' || model.service === 'elasticsearch',
-      isDeployed: deploymentStatsByModelId[trainedModelId] === 'deployed',
+      isDeployable: model.service === Service.elser || model.service === Service.elasticsearch,
```

Comment:
Tried to use [Service](https://github.com/elastic/kibana/blob/14507695ee129e241a4423abf3dfa681bdc4c5c2/x-pack/packages/ml/inference_integration_flyout/types.ts#L54) from` @kbn/inference_integration_flyout`. Please lmk if this needs to be changed